src/UglyToad.PdfPig/Graphics/ReflectionGraphicsStateOperationFactory.cs

```diff
@@ -20,37 +19,86 @@ namespace UglyToad.PdfPig.Graphics
     using System.Linq;
     using System.Reflection;
     using Tokens;
-    using Operations.PathPainting;
 
-    internal class ReflectionGraphicsStateOperationFactory : IGraphicsStateOperationFactory
+    internal sealed class ReflectionGraphicsStateOperationFactory : IGraphicsStateOperationFactory
     {
-        private readonly IReadOnlyDictionary<string, Type> operations;
-
-        public ReflectionGraphicsStateOperationFactory()
-        {
-            var assemblyTypes = Assembly.GetAssembly(typeof(ReflectionGraphicsStateOperationFactory))!.GetTypes();
-
-            var result = new Dictionary<string, Type>();
-
-            foreach (var assemblyType in assemblyTypes)
+        private static readonly IReadOnlyDictionary<string, Type> operations =
```

Comment:
As it stands, ```operations``` is instanced to the class, but as the contents is constant, is it ok for it be static instead?

Reply:
Now that there is a .NET 8 build, it may or may not be useful to see if using a FrozenDictionary would have any benefits here? (just a thought, i don't have any real experience with using those yet)

Reply:
I think that'd be definitely a great use of frozen dictionary here indeed!

Reply:
Updated to use FrozenDictionary in the .NET 8.0 build